lib/Zabbix/Tiny.pm

```diff
@@ -40,45 +40,73 @@ has 'request'         => ( is => 'ro' );
 has 'json_prepared'   => ( is => 'ro' );
 has 'json_executed'   => ( is => 'ro', default => sub { 0 } );
 has 'redo'            => ( is => 'ro' );
+has 'version'         => ( is => 'rw' );
 
 my @content_type = ( 'content-type', 'application/json', );
 
 sub BUILD {
     my $self      = shift;
     my $ua        = $self->ua;
     my $url       = $self->server;
-    my $json_data = {
-        jsonrpc => '2.0',
-        id      => $self->id,
-        method  => 'user.login',
-        params  => {
-            user     => $self->user,
-            password => $self->password,
-        },
-    };
     if ( $self->verify_hostname == 0 ) {
         $ua->ssl_opts( verify_hostname => 0 );
     }
 
     if ( $self->ssl_opts ) {
         $ua->ssl_opts( %{ $self->{ssl_opts} } );
     }
+
+    if ( !$self->version ) {
+        my $json_data = {
+            jsonrpc => '2.0',
+            id      => $self->id,
+            method  => 'apiinfo.version',
+            params  => {
+            },
+        };
+        my $encoded_json = encode_json ($json_data);
+        my $post_response = $ua->post( $self->server, @content_type,
+        Content => $encoded_json);
+        my $response_content = decode_json( $post_response->{_content} );
+        if ( $response_content->{error} ) {
+            my $error = $response_content->{error}->{data};
+            croak("Error: $error");
+        }
+        else {
+            my @version = split(/\./, $response_content->{'result'});
+            $self->{version} = $version[0];
+        }
```

Comment:
Logic looks good, but maybe split to a separate sub to get the version? Also consider using `majorVersion.minorVersion` instead of just the major version for more clarity?  I don't recall Zabbix numbering schema off-hand, but would it make sense to zero pad in the patch/revision number as well?

Reply:
Will step away from using simple string and instead will store value as version object. Which should allow to store major.minor.revision information in easy to read form + allow easly compare it to target version when needed.